src/tui.rs

```diff
@@ -2410,8 +2410,13 @@ impl<'a> Context<'a, '_> {
                         // If the line has some indentation and the user pressed Home,
                         // the first time it'll stop at the indentation. The second time
                         // they press it, it'll move to the true start of the line.
-                        let indent_end = tb.indent_end_logical_pos();
-                        if logical_after.x == 0 && logical_before > indent_end {
+                        //
+                        // If the cursor is already at the start of the line,
+                        // we move it back to the end of the indentation.
+                        if logical_after.x == 0
+                            && let indent_end = tb.indent_end_logical_pos()
+                            && (logical_before > indent_end || logical_before.x == 0)
```

Comment:
Because I was curious about it, I tried implementing it myself. The correct approach is checking for `logical_before.x == 0`.

I also made the optimization I described above, by moving the `indent_end_logical_pos` call into the let chain.